target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -3,200 +3,12 @@
-		;;
-	extreme-networks,ws-ap3805i)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(mtd_get_mac_ascii cfg1 RADIOADDR0)
-		;;
-	glinet,gl-ar750)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary art 0x0) 1)
-		;;
-	iodata,wn-ac1167dgr|\
-	iodata,wn-ac1600dgr2|\
-	sitecom,wlr-7100|\
-	zyxel,nbg6616)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_ascii u-boot-env ethaddr) 1)
-		;;
-	nec,wg800hp)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(mtd_get_mac_text board_data 0x880)
-		;;
-	ocedo,koala|\
-	ocedo,ursus)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(mtd_get_mac_binary art 0xc)
-		;;
-	openmesh,a40|\
-	openmesh,a60|\
-	openmesh,mr1750-v1|\
-	openmesh,mr1750-v2|\
-	openmesh,om5p-ac-v2)
-		caldata_extract "ART" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) 16)
-		;;
-	qihoo,c301)
-		caldata_extract "radiocfg" 0x5000 0x844
-		ath10k_patch_mac $(mtd_get_mac_ascii devdata wlan5mac)
-		;;
-	tplink,archer-a7-v5|\
-	tplink,archer-c2-v3|\
-	tplink,archer-c7-v4|\
-	tplink,archer-c7-v5|\
-	tplink,archer-c25-v1|\
-	tplink,tl-wr902ac-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary info 0x8) -1)
-		;;
-	tplink,archer-c5-v1|\
-	tplink,tl-wdr7500-v3)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary u-boot 0x1fc00) -1)
-		;;
-	tplink,archer-d50-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary romfile 0xf100) 2)
-		;;
-	tplink,archer-d7-v1|\
-	tplink,archer-d7b-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary romfs 0xf100) 2)
-		;;
-	tplink,re350k-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary info 0x8) 2)
-		;;
-	tplink,re355-v1|\
-	tplink,re450-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) -2)
-		;;
-	tplink,re450-v2|\
-	tplink,re450-v3|\
-	tplink,re455-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary info 0x8) 1)
-		;;
-	tplink,tl-wpa8630-v1)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary u-boot 0x0fc00) 1)
-		;;
-	esac
-	;;
-"ath10k/cal-pci-0000:01:00.0.bin")
-	case $board in
-	openmesh,om5p-ac-v1)
-		caldata_extract "ART" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) +16)
-		;;
-	sitecom,wlr-8100)
-		caldata_extract "art" 0x5000 0x844
-		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_ascii u-boot-env ethaddr) 1)
-		;;
-	esac
-	;;
 "ath10k/pre-cal-pci-0000:00:00.0.bin")
 	case $board in
-	asus,rp-ac51|\
-	yuncore,a782|\
-	yuncore,xd4200)
-		caldata_extract "art" 0x5000 0x2f20
-		ln -sf /lib/firmware/ath10k/pre-cal-pci-0000\:00\:00.0.bin \
```

Comment:
Any idea what is the usage of this link? did we use the same calibration for precal and cal?

Reply:
This link seems a copy from ath10k-ct-firmware package. They are useless here.
https://github.com/openwrt/openwrt/blob/920414ca8848fe1b430e436207b4f8c927819368/package/firmware/ath10k-ct-firmware/Makefile#L594-L620

I think those ct-firmware packages need some fix because it will create non-exist soft link if device use nvmem calibration data. And `board-2.bin` seems also useless for ath79 target. I remember only ipq401x devices require it. If I'm wrong please correct me.
```
root@OpenWrt:~# ls -al /lib/firmware/ath10k/QCA9888/hw2.0/
drwxr-xr-x    2 root     root            73 Jan 28 10:10 .
drwxr-xr-x    3 root     root            28 Jan 28 10:10 ..
-rw-r--r--    1 root     root        206444 Jan 28 10:10 board-2.bin
lrwxrwxrwx    1 root     root            30 Jan 28 10:10 board.bin -> ../../cal-pci-0000:01:00.0.bin
-rw-r--r--    1 root     root        630784 Jan 28 10:10 firmware-5.bin
root@OpenWrt:~# cd /lib/firmware/ath10k/QCA9888/hw2.0/
root@OpenWrt:/lib/firmware/ath10k/QCA9888/hw2.0# cat '../../cal-pci-0000:01:00.0.bin'
cat: can't open '../../cal-pci-0000:01:00.0.bin': No such file or directory
```

Reply:
> did we use the same calibration for precal and cal?

In this script, at least they did the same thing. Read cal/pre-cal data from MTD and then save .bin file to the `lib/firmware/ath10k` folder. The `ln` command for ath10k wave2 devices did not really take effect.

Reply:
Ok ath10k fw load is complex enough to make some confusion but from what I notice in the code:
- pre-cal IS OPTIONAL, is loaded from nvmem or tried to be fetched from firmware directory following `pre-cal-%s-%s.bin` 
- cal IS MANDATORY and probe will fail without it. We have board-2.bin

It seems the link was used just to handle the cal file being mandatory for device that didn't had board-2 by using the pre-cal as board file. (on these device pre-cal is API 1)

I guess that is not the case anymore and we always ship board-2.bin on any device that use ath10k correct?